src/plone/rest/traverse.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+from Products.CMFPlone.interfaces.siteroot import IPloneSiteRoot
+from ZPublisher.BaseRequest import DefaultPublishTraverse
+from plone.rest.interfaces import IAPIRequest
+from zope.component import adapts
+from zope.component import queryMultiAdapter
+from zope.interface import implements
+from zope.publisher.interfaces.browser import IBrowserPublisher
+
+NAME_PREFIX = u'rest_'
+
+
+class RESTTraverse(DefaultPublishTraverse):
+    adapts(IPloneSiteRoot, IAPIRequest)
+
+    def publishTraverse(self, request, name):
+        try:
+            obj = super(RESTTraverse, self).publishTraverse(request, name)
+        except KeyError:
```

Comment:
Yes, it's by intention. We only want to catch exceptions that might occur when looking up a named service. Traversal is trying to lookup an item by key (`item = folder[key]`) which raises a `KeyError` if the item does not exist. However for non-folderish items a lookup by key results in a `AttributeError: '__getitem__'`. We can safely assume that the portal object is folderish and lookups by key will only raise `KeyErrors`.